pyServer/AzureDiskInspectService.py

```diff
@@ -213,47 +233,133 @@ def ParseUrlArguments(self, urlPath, sasKey):
+    def ValidateAndParseBlobSasUrl(self, blobSasUrl):
+        self.telemetryLogger.info("Validating input destination Blob Sas Url.")
+
+        urlParts = urllib.parse.urlparse(blobSasUrl)
+
+        if not urlParts.path or len(urlParts.path) == 0:
+            raise InvalidBlobSasUrlException('Input Blob SAS Url for upload is missing Container and Blob info.')
+        if not urlParts.query or len(urlParts.query) == 0:
+            raise InvalidBlobSasUrlException('Input Blob SAS url for upload is missing SAS token.')
+
+        if urlParts.path[0] == '/':
+            blob_storage_path = urlParts.path[1:]
+        else:
+            blob_storage_path = urlParts.path
+
+        if 'blob' not in urlParts.netloc:
+            raise InvalidBlobSasUrlException('Input Blob SAS url for upload is not pointing to a valid Azure Blob endpoint.')
+
+        self.destination_storage_account = urlParts.netloc.split('.')[0]
+
+        if not self.destination_storage_account:
+            raise InvalidBlobSasUrlException('Input Blob SAS url for upload does not contain storage account.')
+
+        urlSplit = blob_storage_path.split('/')
+
+        if not len(urlSplit) == 2 or not urlSplit[0] or not urlSplit[1]:
+            raise InvalidBlobSasUrlException('Input Blob SAS Url is not in the right format.')
+        
+        self.destinaion_container_name = urlSplit[0]
+        self.destinaion_blob_name = urlSplit[1]
+        self.destination_sas_token = urlParts.query
+
+        self.telemetryLogger.info("Destination Blob Storage Account: " + self.destination_storage_account)
+        self.telemetryLogger.info("Container Name: " + self.destinaion_container_name)
+        self.telemetryLogger.info("Blob Name: " + self.destinaion_blob_name)
+
     """
-    def uploadFile(self, http_headers, outputFileName, isPartial, osType):
+    Upload the given file to destination Blob storage using the given Sas Url.
+    """
+    def uploadFileWithBlobSasUrl(self, file_name_full_path):
+        try:
+            sas_service = BlockBlobService(account_name=self.destination_storage_account, sas_token=self.destination_sas_token)
+            self.telemetryLogger.info('Uploading to Blob starting.')
+            start_time = datetime.now()
+            sas_service.create_blob_from_path(self.destinaion_container_name, self.destinaion_blob_name, file_name_full_path)
+            self.telemetryLogger.info('Uploading to Blob completed. Time take: ' + str((datetime.now() - start_time).total_seconds() * 1000) + ' ms')
+        except Exception as ex:
+            raise BlobUploadException(ex)
+
+    """
+    Upload a local file either as a HTTP binary response or upload to a given destination blob storage.
+    """
+    def uploadFile(self, http_headers, outputFileName, isPartial, osType, blobSasUrl):
+
+        if blobSasUrl:
+            self.telemetryLogger.info('Uploading result to Blob storage.')
+            self.uploadFileWithBlobSasUrl(file_name_full_path=outputFileName)
```

Comment:
Prior to this `avoidSendingKeepAlive` is set to true to avoid sending HTTP-100's while we are sending the zip payload in the response.  Since we are not sending the payload but we expect the caller to wait, we should consider if we need to be sending keep-alives while we upload to storage so it doesn't timeout/disconnect.

Reply:
Updated PR to leave keep alive flowing when we upload result to blob.